### PR TITLE
Use `patch` utility for patching sources on all platforms

### PR DESCRIPTION
**Pull request by sergei-mironov**
Let me propose a fix for https://github.com/tensorflow/tensorflow/issues/31196 issue

### CODE REVIEW

**Review by mihaimaruseac — CHANGES_REQUESTED** [review]
This PR does more than its description says it does. Please remove unrelated changes.

**Comment by grwlf** [comment]
> This PR does more than its description says it does. Please remove unrelated changes.

Rebased to master. @mihaimaruseac @rthadur please check

**Review by mihaimaruseac — APPROVED** [review]

File: third_party/repo.bzl [hunk 1]
```diff
@@ -63,10 +63,7 @@ def _repos_are_siblings():
 # Apply a patch_file to the repository root directory
 # Runs 'git apply' on Unix, 'patch -p1' on Windows.
```
Comment (mihaimaruseac) [inline third_party/repo.bzl]: Please also edit this comment.

**Review by mihaimaruseac — CHANGES_REQUESTED** [review]

File: third_party/repo.bzl (hunk 1 above)
Comment (grwlf) [reply third_party/repo.bzl]: Done

**Review by mihaimaruseac — APPROVED** [review]

**Pull request merged**